### DISCUSSION TITLE
Device disabled, device 'adalight' signals error: 'Timeout writing data to ttyACM0'

### DISCUSSION THREAD

**Discussion by versionDefect** [awawa-dev/HyperSerialPico#31] [Q&A]
Hi all. 

I've moved from my ESP32 -> Adafruit Feather RP2040 Ada 4884 and for some reason I'm getting an error when trying to enable the device in HyperHDR. 
logs: https://pastebin.com/DaHQVNxp
Adafruit_Feather_RP2040_Scorpio.zip specifically HyperSerialPico_sk6812Neutral.uf2 is what I used. 

Don't really know what I'm doing wrong here. What do you guys think?

**Comment by awawa-dev** [comment]
Hi
> 2024-07-16T06:24:10.776Z [LEDDEVICE_ADALIGHT] Could not detect **HyperSerialEsp8266/HyperSerialESP32** device

You must use HyperHDR v20 

```
HyperHDR configuration
You need HyperHDR v20 or newer. Make sure you have enabled AWA protocol and Esp8266/ESP32/Rp2040 handshake options.
```

**Reply by versionDefect** [reply]
Damn sometimes I feel like I can't read. Thanks. While I've got you here, I've plugged everything in, soldered the pins so i don't have any werid screwyness going on. But only the NeoPixel on the board is turning on and nothing on the LEDs themselves. I've of course connected (-) to GND on the board and GPIO 2 to the LED data pin but still nothing for some reason. 
https://pastebin.com/v0tZecbL (that reset was me btw so you can safely ignore that)

**Reply by awawa-dev** [reply]
GPIO2 is wrong for Adafruit boards firmware. They need different pinout because of the built-in level shifter. The manual mentions about it: check Adafruit_Feather_RP2040_Scorpio.zip for pinout.

**Reply by versionDefect** [reply]
> GPIO2 is wrong for Adafruit boards firmware. They need different pinout because of the built-in level shifter. The manual mentions about it: check Adafruit_Feather_RP2040_Scorpio.zip for pinout.

Ahh i see. Looking at it now I see the documentation you've left. 

Neopixel is using GPIO16(OUTPUT_DATA_PIN) on output 0.
SPI is using spi0 interface pins: GPIO19(OUTPUT_SPI_DATA_PIN) and GPIO18(OUTPUT_SPI_CLOCK_PIN) on output 3 and 2 respectively.

Gonna knock out it's late so i'll fix that tomorrow. Thanks for all your help :-). 

ps, just to make sure as a sanity check i'd just use GPIO19 right?

**Reply by awawa-dev** [reply]
Neopixel is using GPIO16(OUTPUT_DATA_PIN) on output 0.

**Reply by versionDefect** [reply]
> Neopixel is using GPIO16(OUTPUT_DATA_PIN) on output 0.

Oh, I assumed Neopixel had it reserved, So it would share the same pin? Got it. Thanks for the clarification

**Reply by versionDefect** [reply]
> > Neopixel is using GPIO16(OUTPUT_DATA_PIN) on output 0.
> 
> Oh, I assumed Neopixel had it reserved, So it would share the same pin? Got it. Thanks for the clarification 

Thinking about it now it does make sense why it does that. Sorry very sleep deprived right now haha.
